Common/src/grid_movement/CFreeFormDefBox.cpp

```diff
@@ -108,13 +108,17 @@ CFreeFormDefBox::~CFreeFormDefBox(void) {
         delete [] Coord_Control_Points[iOrder][jOrder][kOrder];
         delete [] ParCoord_Control_Points[iOrder][jOrder][kOrder];
         delete [] Coord_Control_Points_Copy[iOrder][jOrder][kOrder];
+        if (Coord_SupportCP != nullptr) delete [] Coord_SupportCP[iOrder][jOrder][kOrder];
```

Comment:
Is each of the dimensions allocated with 'new'? If so the middle arrays are being leaked.

Reply:
Ah, I should first `delete [] array[i[j][k]`, then `delete [] array[i][j]` then `delete [] array[i]` ?

Reply:
If all the dimensions are allocated with new, yes, unfortunately.

Reply:
yep, looks like it, also for all the other variables there